internal/infrastructure/kubernetes/proxy/resource_provider.go

```diff
@@ -294,3 +294,10 @@ func (r *ResourceRender) HorizontalPodAutoscaler() (*autoscalingv2.HorizontalPod
 
 	return hpa, nil
 }
+
+// lackOwningGatewayLabels Check if labels are missing some OwningGatewayLabels
+func lackOwningGatewayLabels(labels map[string]string) bool {
```

Comment:
```suggestion
func OwningGatewayLabelsAbsent(labels map[string]string) bool {
```

Reply:
This does indeed seem more suitable, I have made the changes.